### PR TITLE
Fix missing marshallers for apps protobuf and fix verify-generated-protobuf

### LINKED ISSUE #24579: TestRoundTripTypes flake in serialization_test.go: v1alpha1.PetSet does not implement the protobuf marshalling interface
During https://github.com/kubernetes/kubernetes/pull/24519#issuecomment-212688280

https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/24519/kubernetes-pull-test-unit-integration/22150/?pli=1

```sh
FAIL
FAIL	k8s.io/kubernetes/pkg/api	8.295s

--- FAIL: TestRoundTripTypes (6.32s)
	serialization_test.go:165: working on ThirdPartyResourceList in extensions
	serialization_test.go:165: working on Job in extensions
	serialization_test.go:165: working on PodSecurityPolicy in extensions
	serialization_test.go:165: working on ThirdPartyResourceDataList in extensions
	serialization_test.go:165: working on HorizontalPodAutoscalerList in extensions
	serialization_test.go:165: working on ReplicationControllerDummy in extensions
	serialization_test.go:165: working on ListOptions in extensions
	serialization_test.go:165: working on Deployment in extensions
	serialization_test.go:165: working on JobList in extensions
	serialization_test.go:165: working on Scale in extensions
	serialization_test.go:165: working on PodSecurityPolicyList in extensions
	serialization_test.go:165: working on ThirdPartyResource in extensions
	serialization_test.go:165: working on IngressList in extensions
	serialization_test.go:165: working on DeploymentRollback in extensions
	serialization_test.go:165: working on HorizontalPodAutoscaler in extensions
	serialization_test.go:165: working on ThirdPartyResourceData in extensions
	serialization_test.go:165: working on WatchEvent in extensions
	serialization_test.go:165: working on DaemonSet in extensions
	serialization_test.go:165: working on ReplicaSetList in extensions
	serialization_test.go:165: working on ReplicaSet in extensions
	serialization_test.go:165: working on DaemonSetList in extensions
	serialization_test.go:165: working on Ingress in extensions
	serialization_test.go:165: working on DeploymentList in extensions
	serialization_test.go:165: working on ExportOptions in extensions
	serialization_test.go:165: working on PetSet in apps
	serialization_test.go:79: PetSet: object *v1alpha1.PetSet does not implement the protobuf marshalling interface and cannot be encoded to a protobuf message ((*apps.PetSet){TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string)é¦¥é³žå¥“æº«Â®'æ§£è®„Ç§É´é¶šå€¨  GenerateName:(string)Æ‚'ÉŽÂ¢)Ä¦É“Â¬ç¶š)Ç—æ‡‰é› Ç´ÃŸAçŠ¡Ä Namespace:(string) SelfLink:(string)Â´çŠÃ›((È¼TÄ 3u UID:(types.UID)mGÉ†È=\K{<ÇÉ˜ ResourceVersion:(string)13493595933874889695 Generation:(int64)0 CreationTimestamp:(unversioned.Time){Time:(time.Time){sec:(int64)0 nsec:(int32)0 loc:(*time.Location)<nil>}} DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(apps.PetSetSpec){Replicas:(int)921267143 Selector:(*unversioned.LabelSelector){MatchLabels:(map[string]string)map[CÆ°é·šÅ²Æ«é—¨ÉšÅ®ä¼‹Ä•Çç²¬::'å˜‰pÇµé„·É ] MatchExpressions:([]unversioned.LabelSelectorRequirement)[{Key:(string)æ€†] Operator:(unversioned.LabelSelectorOperator)ÄµÂ£&é€eè†‚å©Uç¼Ã”u Values:([]string)[ÄŠÈºÄšNç¸ºé—½Å™Ãªeoæ”•Æ½å‚ZÅƒé½]}]} Template:(api.PodTemplateSpec){ObjectMeta:(api.ObjectMeta){Name:(string) GenerateName:(string)zè—» Namespace:(string) SelfLink:(string)WQR UID:(types.UID)Éœ.Ç¥åª²æ’¾4Ã¬å‚ŽLæˆWæ¸– ResourceVersion:(string)7095254002371897998 Generation:(int64)0 CreationTimestamp:(unversioned.Time){Time:(time.Time){sec:(int64)0 nsec:(int32)0 loc:(*time.Location)<nil>}} DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PodSpec){Volumes:([]api.Volume)<nil> Containers:([]api.Container)[{Name:(string) -æ©·Ç³èµƒ1Ä½å„ŽÂ«Æ«- Image:(string)bæ„«åŽ¥è«œÊ Command:([]string)<nil> Args:([]string)<nil> WorkingDir:(string)Q Ports:([]api.ContainerPort)[{Name:(string)è´¶Æ±æ‹–è´§ÊŒÈCè‚¶é¡¼dåŸŽèŸ¸ HostPort:(int)1371157032 ContainerPort:(int)814508831 Protocol:(api.Protocol)UDP HostIP:(string)å„‘Ã€ Ä€æŸ„}] Env:([]api.EnvVar)<nil> Resources:(api.ResourceRequirements){Limits:(api.ResourceList)map[cpu:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x211]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} memory:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x9d]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} storage:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x1df]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}] Requests:(api.ResourceList)map[cpu:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x211]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} memory:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x9d]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} storage:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x1df]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}]} VolumeMounts:([]api.VolumeMount)<nil> LivenessProbe:(*api.Probe){Handler:(api.Handler){Exec:(*api.ExecAction)<nil> HTTPGet:(*api.HTTPGetAction){Path:(string)/ç˜™Èœæ»‘Mç°²ä¹œæŠ—é™Žçœ»ç—™bGq9ç½·Æ–k Port:(intstr.IntOrString){Type:(intstr.Type)1 IntVal:(int32)0 StrVal:(string)ÅšHCÊŽÂ°ç³˜Ê€v} Host:(string)è¼žè¹¡uKè¹»Ç½Å›Âºè‘—r0 Scheme:(api.URIScheme)xÅ©å¿£%Â¸Ç«kÃ½=Læ•™Æ…å‚§Ãƒ HTTPHeaders:([]api.HTTPHeader)<nil>} TCPSocket:(*api.TCPSocketAction)<nil>} InitialDelaySeconds:(int)2129797302 TimeoutSeconds:(int)1484528591 PeriodSeconds:(int)59358248 SuccessThreshold:(int)1898790883 FailureThreshold:(int)1140915297} ReadinessProbe:(*api.Probe){Handler:(api.Handler){Exec:(*api.ExecAction)<nil> HTTPGet:(*api.HTTPGetAction){Path:(string)/Æ¨Æ€3æ¾è²µé¸„æ‚šÄ¢2É”Ä­æ¨ˆ_æ…—ÆŽlAæˆ Port:(intstr.IntOrString){Type:(intstr.Type)1 IntVal:(int32)0 StrVal:(string)'} Host:(string)[å‹¶7É‡RÆ›Æ˜@l Scheme:(api.URIScheme)xe)Ã¢y)É¶É­)é¾¬æ˜¼å‹Š HTTPHeaders:([]api.HTTPHeader)<nil>} TCPSocket:(*api.TCPSocketAction)<nil>} InitialDelaySeconds:(int)1327025088 TimeoutSeconds:(int)867072522 PeriodSeconds:(int)227967599 SuccessThreshold:(int)1916573673 FailureThreshold:(int)1048123472} Lifecycle:(*api.Lifecycle)<nil> TerminationMessagePath:(string)/éº‹v ImagePullPolicy:(api.PullPolicy)Always SecurityContext:(*api.SecurityContext){Capabilities:(*api.Capabilities){Add:([]api.Capability)[Ê”] Drop:([]api.Capability)<nil>} Privileged:(*bool)<nil> SELinuxOptions:(*api.SELinuxOptions){User:(string)È¿h%é‘è°Œ:Ç‡vé‹˜ Role:(string) Type:(string),xb^é›†AÆ¨3kè‹„ç¡¥èµ«fÇŒÉš Level:(string)mæ«µæœ®å›°[} RunAsUser:(*int64)-1097654023993897482 RunAsNonRoot:(*bool)<nil> ReadOnlyRootFilesystem:(*bool)<nil>} Stdin:(bool)true StdinOnce:(bool)false TTY:(bool)true}] RestartPolicy:(api.RestartPolicy)Never TerminationGracePeriodSeconds:(*int64)1273864270 ActiveDeadlineSeconds:(*int64)<nil> DNSPolicy:(api.DNSPolicy)ClusterFirst NodeSelector:(map[string]string)<nil> ServiceAccountName:(string)ä½§pÈƒÂ¸È Å’æ£¥CqfI NodeName:(string)cÃÄ·pè‰VÈ‚Ä™6 SecurityContext:(*api.PodSecurityContext){HostNetwork:(bool)false HostPID:(bool)false HostIPC:(bool)false SELinuxOptions:(*api.SELinuxOptions)<nil> RunAsUser:(*int64)<nil> RunAsNonRoot:(*bool)<nil> SupplementalGroups:([]int64)<nil> FSGroup:(*int64)<nil>} ImagePullSecrets:([]api.LocalObjectReference)[{Name:(string)å‹œÉ„è¿šä»´ççŽ´fæº‡}]}} VolumeClaimTemplates:([]api.PersistentVolumeClaim)[{TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string) GenerateName:(string)pÂ¸èº¨Âªæ«°ewÆ¤æ¬¾gç¢ÇœÇ™6È«é¾‘æŸ•È¦Æ› Namespace:(string) SelfLink:(string)>ä½šSÃ¯è˜³"Pç§½Æ‹DÈŒæ·‡ UID:(types.UID) ResourceVersion:(string)13400519669593728760 Generation:(int64)0 CreationTimestamp:(unversioned.Time){Time:(time.Time){sec:(int64)0 nsec:(int32)0 loc:(*time.Location)<nil>}} DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PersistentVolumeClaimSpec){AccessModes:([]api.PersistentVolumeAccessMode)[å°‚#é”è§¸åš¶Ç—}Ç‰iÃ¢O4æµŽÇ”ç¾ƒè«„Ç¿É›] Resources:(api.ResourceRequirements){Limits:(api.ResourceList)map[cpu:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x36b]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} memory:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x379]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} storage:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x116]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}] Requests:(api.ResourceList)map[cpu:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x36b]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} memory:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x379]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} storage:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x116]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}]} VolumeName:(string)=å¤³} Status:(api.PersistentVolumeClaimStatus){Phase:(api.PersistentVolumeClaimPhase)Pending AccessModes:([]api.PersistentVolumeAccessMode)<nil> Capacity:(api.ResourceList)map[)Å„Â¦çžÅ¯\ç£¾éŒÅ¶?WBGr å·±Ê¬bå±‘:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0xa8]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}]}}] ServiceName:(string)é©—Ç¥Éž]} Status:(apps.PetSetStatus){ObservedGeneration:(*int64)<nil> Replicas:(int)311351303}})
	serialization_test.go:165: working on WatchEvent in apps
	serialization_test.go:165: working on PetSetList in apps
	serialization_test.go:79: PetSetList: object *v1alpha1.PetSetList does not implement the protobuf marshalling interface and cannot be encoded to a protobuf message ((*apps.PetSetList){TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ListMeta:(unversioned.ListMeta){SelfLink:(string)æ—«V[eÆ½ÆµÇ„Ãjé£§ ResourceVersion:(string)6035297283636688115} Items:([]apps.PetSet)[{TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string)yçµ€K#èŠ¹è®Ã¦ GenerateName:(string)è´‹Ä‹Å› Namespace:(string) SelfLink:(string)é¨ UID:(types.UID)>æ§«0äºœÄ¤\èªªÊ‹Ç  ResourceVersion:(string)15728989767038634282 Generation:(int64)0 CreationTimestamp:(unversioned.Time){Time:(time.Time){sec:(int64)0 nsec:(int32)0 loc:(*time.Location)<nil>}} DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(apps.PetSetSpec){Replicas:(int)742700660 Selector:(*unversioned.LabelSelector){MatchLabels:(map[string]string)<nil> MatchExpressions:([]unversioned.LabelSelectorRequirement)[{Key:(string){ç¥®<é€æŽ…É‘Â£Êˆåš¿å¢¢é¼·ÆµF Operator:(unversioned.LabelSelectorOperator)ÈžÃªç§‘ç¥‹æ·™Ç‹NÄ¼+,å –å¢§Wæ¦æ´šå“‘Äœ<é¥¡ Values:([]string)[éµ¬æ¬\èŸ³zÆ¸é«¢C#å¤‰]}]} Template:(api.PodTemplateSpec){ObjectMeta:(api.ObjectMeta){Name:(string)ÄŸÊŽÈÄ‹Zå˜Åjn3Ä‚Èº GenerateName:(string)U Namespace:(string) SelfLink:(string)L!A=ä¿—83)9S UID:(types.UID)Â¿ÅŽ ResourceVersion:(string)18097595000190335219 Generation:(int64)0 CreationTimestamp:(unversioned.Time){Time:(time.Time){sec:(int64)0 nsec:(int32)0 loc:(*time.Location)<nil>}} DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PodSpec){Volumes:([]api.Volume)[{Name:(string)Ã“ÄÂ°Ã´å–™è…Œ@Ã¬å•‰åŽ¸Ä¬ Ã¿è›§çž¶é‘°'\? VolumeSource:(api.VolumeSource){HostPath:(*api.HostPathVolumeSource)<nil> EmptyDir:(*api.EmptyDirVolumeSource)<nil> GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)<nil> AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSource)<nil> GitRepo:(*api.GitRepoVolumeSource){Repository:(string)Ê`='Ttç¸€ä¸˜ Revision:(string)åœ½ÇŠFoÇˆ Directory:(string)lZå— Çµèƒ¿=é‡ƒå˜å’€é‡±e} Secret:(*api.SecretVolumeSource)<nil> NFS:(*api.NFSVolumeSource)<nil> ISCSI:(*api.ISCSIVolumeSource)<nil> Glusterfs:(*api.GlusterfsVolumeSource)<nil> PersistentVolumeClaim:(*api.PersistentVolumeClaimVolumeSource)<nil> RBD:(*api.RBDVolumeSource)<nil> FlexVolume:(*api.FlexVolumeSource)<nil> Cinder:(*api.CinderVolumeSource)<nil> CephFS:(*api.CephFSVolumeSource)<nil> Flocker:(*api.FlockerVolumeSource)<nil> DownwardAPI:(*api.DownwardAPIVolumeSource)<nil> FC:(*api.FCVolumeSource)<nil> AzureFile:(*api.AzureFileVolumeSource)<nil> ConfigMap:(*api.ConfigMapVolumeSource)<nil>}}] Containers:([]api.Container)<nil> RestartPolicy:(api.RestartPolicy)Never TerminationGracePeriodSeconds:(*int64)30 ActiveDeadlineSeconds:(*int64)<nil> DNSPolicy:(api.DNSPolicy)Default NodeSelector:(map[string]string)<nil> ServiceAccountName:(string)jBå¦œÆ©å¸ÊŒåäº›Æé•€ÇÂ¦$oÆ³dÄ‡ç— NodeName:(string)æ„µè¾†È¹iÇŽç¤µÄ¥å­¯Æ²ç©½Y%Æ£Ê’Ã‘Ã‰ SecurityContext:(*api.PodSecurityContext){HostNetwork:(bool)true HostPID:(bool)false HostIPC:(bool)false SELinuxOptions:(*api.SELinuxOptions)<nil> RunAsUser:(*int64)<nil> RunAsNonRoot:(*bool)<nil> SupplementalGroups:([]int64)<nil> FSGroup:(*int64)3560504669839892368} ImagePullSecrets:([]api.LocalObjectReference)<nil>}} VolumeClaimTemplates:([]api.PersistentVolumeClaim)[{TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string)Træ£É¹e GenerateName:(string)Å§< Namespace:(string) SelfLink:(string)EÇŒ UID:(types.UID)Å¾Ä•ç“–SpkÈ’Ä…È´Ê¬É·é§é›® ResourceVersion:(string)12592203241854516216 Generation:(int64)0 CreationTimestamp:(unversioned.Time){Time:(time.Time){sec:(int64)0 nsec:(int32)0 loc:(*time.Location)<nil>}} DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PersistentVolumeClaimSpec){AccessModes:([]api.PersistentVolumeAccessMode)<nil> Resources:(api.ResourceRequirements){Limits:(api.ResourceList)map[cpu:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x336]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} memory:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x123]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} storage:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x365]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}] Requests:(api.ResourceList)map[memory:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x123]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} storage:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x365]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent} cpu:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x336]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}]} VolumeName:(string)} Status:(api.PersistentVolumeClaimStatus){Phase:(api.PersistentVolumeClaimPhase)Bound AccessModes:([]api.PersistentVolumeAccessMode)[pVÂ¿ysé…µÂ§Å¯é¯ŠÇ·v6É´ÈŽ] Capacity:(api.ResourceList)map[å‡¸Ä»å¨¼ÄƒÈ—}=Æ¤Æ¿æ’²å¥…/:{Amount:(*inf.Dec){unscaled:(big.Int){neg:(bool)false abs:(big.nat)[0x1e]} scale:(inf.Scale)0} Format:(resource.Format)DecimalExponent}]}}] ServiceName:(string)S4éƒ‰)Ê‡$cè¯ª} Status:(apps.PetSetStatus){ObservedGeneration:(*int64)8488688865181050076 Replicas:(int)523717373}}]})
	serialization_test.go:165: working on ListOptions in apps
```

### PR DESCRIPTION
**Pull request by smarterclayton**
Fixes #24579

### CODE REVIEW

**Comment by smarterclayton** [comment]
@k8s-oncall merge queue is broken by bad verify script (that didn't catch a new required gap) - this should fix it (although I may have to disable the new check on travis)

**Comment by smarterclayton** [comment]
@wojtek-t update-generated-protobuf failed remotely but worked locally:

```
Verifying ./hack/../hack/verify-generated-protobuf.sh
+++ [0420 22:25:02] Verifying Prerequisites....
+++ [0420 22:25:02] Building Docker image kube-build:build-91dbedbcaf.
+++ [0420 22:25:06] Running build command....
+++ [0420 22:25:06] Creating data container
exec: "hack/after-build/update-generated-protobuf.sh": stat hack/after-build/update-generated-protobuf.sh: no such file or directory
Error response from daemon: Cannot start container 0a3b839e0e7d9b3b0caddc9817fa2d2b6b2096708187370af1d4af89b3bfd638: [8] System error: exec: "hack/after-build/update-generated-protobuf.sh": stat hack/after-build/update-generated-protobuf.sh: no such file or directory
!!! Error in /go/src/k8s.io/kubernetes/hack/../build/common.sh:594
  '"${docker_cmd[@]}" "$@"' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/../build/common.sh:594 kube::build::run_build_command(...)
  2: /go/src/k8s.io/kubernetes/hack/update-generated-protobuf.sh:52 main(...)
Exiting with status 1
!!! Error in /go/src/k8s.io/kubernetes/hack/after-build/verify-generated-protobuf.sh:39
  '"${KUBE_ROOT}/hack/update-generated-protobuf.sh"' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/after-build/verify-generated-protobuf.sh:39 main(...)
Exiting with status 1
!!! Error in ./hack/../hack/verify-generated-protobuf.sh:41
  '"${KUBE_ROOT}/hack/after-build/verify-generated-protobuf.sh" "$@"' exited with status 1
Call stack:
  1: ./hack/../hack/verify-generated-protobuf.sh:41 main(...)
Exiting with status 1
FAILED   ./hack/../hack/verify-generated-protobuf.sh	8s
```

Something about jenkins environment that we never verified?

**Comment by smarterclayton** [comment]
Possibly the docker build image on the jenkins worker contains other content, or a path is wrong?

**Comment by smarterclayton** [comment]
@wojtek-t also

```
cp: cannot create regular file ‘/home/travis/gopath/src/github.com/kubernetes/kubernetes/pkg/api/v1/generated.pb.go’: Permission denied
cp: cannot create regular file ‘/home/travis/gopath/src/github.com/kubernetes/kubernetes/pkg/api/v1/generated.proto’: Permission denied
cp: cannot create regular file ‘/home/travis/gopath/src/github.com/kubernetes/kubernetes/pkg/api/resource/generated.pb.go’: Permission denied
cp: cannot create regular file ‘/home/travis/gopath/src/github.com/kubernetes/kubernetes/pkg/api/resource/generated.proto’: Permission denied
cp: cannot create regular file ‘/home/travis/gopath/src/github.com/kubernetes/kubernetes/pkg/api/unversioned/generated.pb.go’: Permission denied
cp: cannot create regular file ‘/home/travis/gopath/src/github.com/kubernetes/kubernetes/pkg/api/unversioned/generated.proto’: Permission denied
!!! Error in /home/travis/gopath/src/github.com/kubernetes/kubernetes/hack/after-build/verify-generated-protobuf.sh:45
 ```

from travis.  Not expected

File: hack/after-build/update-generated-protobuf.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ if [[ -z "$(which protoc)" || "$(protoc --version)" != "libprotoc 3.0."* ]]; the
   echo "  https://github.com/google/protobuf/releases"
   echo
   echo "WARNING: Protobuf changes are not being validated"
-  exit 0
+  exit 1
```
Comment (wojtek-t) [inline hack/after-build/update-generated-protobuf.sh]: yeah - I should have caught that...

**Comment by wojtek-t** [comment]
Hmm - I don't know much about Jenkins.
@ixdy @spxtr @fejta - any thoughts what is happening here?

**Comment by ixdy** [comment]
On Jenkins, we run the verification, unit, and integration checks inside the `kubekins-test` docker image. We support docker-in-docker, but you have to use the host path for volume mounts, not the docker container path. Thus you're probably failing to mount the right volumes in `hack/update-generated-protobuf.sh`.

Luckily, we already have some checks that have to do this, so we already have an environment variable pointing to the host's repository directory - see https://github.com/kubernetes/kubernetes/blob/master/hack/update-api-reference-docs.sh#L32-L38. Hopefully that's enough.

**Comment by ixdy** [comment]
tl;dr in `hack/update-generated-protobuf.sh`, set `KUBE_ROOT` to `REPO_DIR` if the latter is set, and see if everything suddenly works on Jenkins.

File: hack/verify-generated-protobuf.sh [hunk 2]
```diff
@@ -19,11 +19,24 @@ set -o nounset
 set -o pipefail
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
-source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/build/common.sh"
 
 kube::golang::setup_env
 
-"${KUBE_ROOT}/hack/build-go.sh" cmd/libs/go2idl/go-to-protobuf cmd/libs/go2idl/go-to-protobuf/protoc-gen-gogo
+function prereqs() {
```
Comment (wojtek-t) [inline hack/verify-generated-protobuf.sh]: I vote for removing this function - this is done anyway as part of update-generated-protobuf, so duplicating this logic doesn't make much sense.

File: pkg/apis/apps/v1alpha1/generated.pb.go [hunk 3]
```diff
@@ -0,0 +1,971 @@
+/*
```
Comment (wojtek-t) [inline pkg/apis/apps/v1alpha1/generated.pb.go]: Also - can you please rebase - these files were already merged as part of a different commit.

**Comment by wojtek-t** [comment]
@ixdy - thanks!
Verifying now.

**Comment by wojtek-t** [comment]
I'm experimenting with it in https://github.com/kubernetes/kubernetes/pull/24594 - but so far without success.

**Comment by wojtek-t** [comment]
OK - I'm a step further. Hopefully will fix that soon.

**Comment by wojtek-t** [comment]
So switching to REPO_DIR (in a correct place, which I didn't do the first time) helps with the first problem.
However, there is still permissions issue, and I have the same issue locally actually.

**Comment by wojtek-t** [comment]
@ixdy - thanks for pointing to that; it actually fixed the problem

**Comment by wojtek-t** [comment]
I'm few steps further in https://github.com/kubernetes/kubernetes/pull/24594
So I'm closing this one in favor of that PR. Hopefully that's not a problem.
